pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ComplexTypeTransformer.java

```diff
@@ -145,6 +149,21 @@ public static ComplexTypeTransformer getComplexTypeTransformer(TableConfig table
     }
   }
 
+  public static Set<String> getFieldsToReadWithComplexType(Set<String> fieldsToRead, IngestionConfig ingestionConfig) {
+    if (ingestionConfig == null || ingestionConfig.getComplexTypeConfig() == null) {
+      // do nothing
+      return fieldsToRead;
+    }
+    ComplexTypeConfig complexTypeConfig = ingestionConfig.getComplexTypeConfig();
+    Set<String> result = new HashSet<>();
+    String delimiter = complexTypeConfig.getDelimiter() == null ? DEFAULT_DELIMITER : complexTypeConfig.getDelimiter();
+    for (String field : fieldsToRead) {
+      // need to quote because split takes regex as arg
+      result.add(field.split(Pattern.quote(delimiter))[0]);
```

Comment:
Use `org.apache.commons.lang3.StringUtils.split()` which does not do regex matching